asg-passes/src/dotify/dotgraph.rs

```diff
@@ -259,52 +231,30 @@ mod tests {
     #[test]
     fn test_render_reachable_set() -> Result<(), Box<dyn Error>> {
         let mut graph = DotGraph::new("example1".to_string());
-        let node0 = graph.add_node(DotNode::new("N0".to_string(), "".to_string()));
-        let node1 = graph.add_node(DotNode::new("N1".to_string(), "".to_string()));
-        let node2 = graph.add_node(DotNode::new("N2".to_string(), "".to_string()));
-        let node3 = graph.add_node(DotNode::new("N3".to_string(), "".to_string()));
-        let node4 = graph.add_node(DotNode::new("N4".to_string(), "".to_string()));
-
-        let edge0 = DotEdge {
-            start_idx: node0,
-            end_idx: node1,
-            label: "".to_string(),
-            color: "black",
-        };
-
-        let edge2 = DotEdge {
-            start_idx: node1,
-            end_idx: node3,
-            label: "".to_string(),
-            color: "black",
-        };
-
-        let edge3 = DotEdge {
-            start_idx: node2,
-            end_idx: node3,
-            label: "".to_string(),
-            color: "black",
-        };
-
-        let edge4 = DotEdge {
-            start_idx: node3,
-            end_idx: node4,
-            label: "".to_string(),
-            color: "black",
-        };
 
-        let edge5 = DotEdge {
-            start_idx: node4,
-            end_idx: node4,
-            label: "".to_string(),
-            color: "black",
+        let mut add_node = |id: &str| graph.add_node(DotNode::new(id.to_string(), "".to_string()));
+
+        let node0 = add_node("N0");
+        let node1 = add_node("N1");
+        let node2 = add_node("N2");
+        let node3 = add_node("N3");
+        let node4 = add_node("N4");
+
+        let mut add_edge = |start_idx, end_idx| {
+            let edge = DotEdge {
+                start_idx,
+                end_idx,
+                label: "".to_string(),
+                color: "black",
+            };
```

Comment:
Btw, consider extracting this bit to a top level function since its repeated twice so you get `graph.add_edge(edge(start, end));`